### PR TITLE
FP-23 added tests for ProductsController

### PR DESCRIPTION
**Pull request by natam**
FP-23 added tests for ProductsController

### CODE REVIEW

File: src/main/java/com/nkh/ECommerceShop/exception/ControllerExceptionHandler.java [hunk 1]
```diff
@@ -31,8 +31,8 @@ public ResponseEntity<ErrorMessage> resourceNotFoundException(ResourceNotFoundEx
         return new ResponseEntity<>(message, HttpStatus.NOT_FOUND);
     }
 
-    @ExceptionHandler(AlreadyExistsException.class)
-    public ResponseEntity<ErrorMessage> resourceNotFoundException(AlreadyExistsException ex, WebRequest request) {
+    @ExceptionHandler({AlreadyExistsException.class})
```
Comment (moseseth) [inline src/main/java/com/nkh/ECommerceShop/exception/ControllerExceptionHandler.java:34]: Why you added the curly brace around the Exception class, that is used for multiple classes?!

File: src/main/java/com/nkh/ECommerceShop/service/ProductsService.java [hunk 2]
```diff
@@ -40,7 +40,8 @@ public boolean checkProductStock(long productId, int requestedAmount){
     }
 
     @Transactional
-    public void deleteProduct(long id){
+    public boolean deleteProduct(long id){
```
Comment (moseseth) [inline src/main/java/com/nkh/ECommerceShop/service/ProductsService.java:43]: What is the reason behind using transactional here as you have only delete operation?

File: src/main/java/com/nkh/ECommerceShop/exception/ControllerExceptionHandler.java (hunk 1 above)
Comment (natam) [reply src/main/java/com/nkh/ECommerceShop/exception/ControllerExceptionHandler.java]: fixed. just had an issue - played a bit with different solutions and forgot to remove.

File: src/main/java/com/nkh/ECommerceShop/service/ProductsService.java (hunk 2 above)
Comment (natam) [reply src/main/java/com/nkh/ECommerceShop/service/ProductsService.java:43]: 2: get by id and delete

**Review by moseseth — APPROVED** [review]

**Pull request merged**